RELEASE_CHECKLIST.md

```diff
@@ -2,18 +2,23 @@
 
 ## Overview
 
-This document describes the checklist to publish a release via GitHub workflow.
-
-NOTE: Make sure the dependencies in `go.mod` file are expected by the release. For example, if there are dependencies on certain version of notation library (notation-go or notation-core-go) or ORAS library (oras-go), make sure that version of library is released first, and the version number is updated accordingly in `go.mod` file.  After updating go.mod file, run `go mod tidy` to ensure the go.sum file is also updated with any potential changes.
+This document describes the checklist to publish a release for Notation CLI via GitHub workflow.

```

Comment:
Since this is a release checklist, under the `Release Process` session, we should add a new section for branch/patch releases, i.e., when there's new features in `main` but we just want a `patch` release.

Reply:
created a new issue for tracking this comment #985